packages/flutter_tools/lib/src/commands/start.dart

```diff
@@ -132,10 +132,9 @@ class StartCommand extends StartCommandBase {
   Future<int> runInProject() async {
     logging.fine('Downloading toolchain.');
 
-    await Future.wait([
-      downloadToolchain(),
-      downloadApplicationPackagesAndConnectToDevices(),
-    ]);
+    // Load these in sequence so that errors common to them aren't printed twice.
```

Comment:
I'll switch it back to use `Future.wait` and pass in the `eagerError: true`. The first failure will complete the overall Future.